packages/api/src/auth/decoders/azureActiveDirectory.ts

```diff
@@ -0,0 +1,49 @@
+import jwt from 'jsonwebtoken'
+import jwksClient from 'jwks-rsa'
+
+export const verifyAzureActiveDirectoryToken = (
+  bearerToken: string
+): Promise<null | Record<string, unknown>> => {
+  return new Promise((resolve, reject) => {
+    const { AZURE_ACTIVE_DIRECTORY_AUTHORITY } = process.env
+    if (!AZURE_ACTIVE_DIRECTORY_AUTHORITY) {
+      throw new Error(
+        '`AZURE_ACTIVE_DIRECTORY_AUTHORITY` env vars are not set.'
+      )
+    }
+
+    // @docs:   https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#sample-response
+    const client = jwksClient({
+      jwksUri: `${AZURE_ACTIVE_DIRECTORY_AUTHORITY}/discovery/v2.0/keys`,
+    })
+
+    jwt.verify(
+      bearerToken,
+      (header, callback) => {
+        client.getSigningKey(header.kid as string, (error, key) => {
+          callback(error, key.getPublicKey())
+        })
+      },
+      {
+        issuer: `${AZURE_ACTIVE_DIRECTORY_AUTHORITY}/v2.0`,
+        algorithms: ['RS256'],
+      },
+      (verifyError, decoded) => {
+        if (verifyError) {
+          return reject(verifyError)
+        }
+        resolve(
+          typeof decoded === 'undefined'
+            ? null
+            : (decoded as Record<string, unknown>)
+        )
+      }
+    )
+  })
+}
+
+export const azureActiveDirectory = async (
+  token: string
+): Promise<null | Record<string, unknown>> => {
+  return verifyAzureActiveDirectoryToken(token)
```

Comment:
Any reason you didn't just inline all the code from that function?

Reply:
I took a shortcut of using [Auth0 as implementation example](https://github.com/redwoodjs/redwood/blob/main/packages/api/src/auth/decoders/auth0.ts#L25-L70). I think we can leave it for now, and revisit that with some refactoring on both auth providers in a separate PR.